test/SdkUnitTests.js

```diff
@@ -1295,4 +1296,344 @@ describe('SDK Unit Tests:', function (done) {
+
+    envelopesApi.getEnvelope(accountId, envelopeId)
+      .then(function (envelope) {
+        assert.notStrictEqual(envelope, undefined);
+        done();
+      })
+      .catch(function (error) {
+        if (error) {
+          return done(error);
+        }
+      });
+  });
+  
+  it('Get envelope recipients', function(done) {
+    var envelopesApi = new docusign.EnvelopesApi(apiClient);
+
+    envelopesApi.listRecipients(accountId, envelopeId)
+      .then(function (recipients) {
+        assert.notStrictEqual(recipients, undefined);
+        done();
+      })
+      .catch(function (error) {
+        if (error) {
+          return done(error);
+        }
+      });      
+  })
+
+  it('Create sender view', function(done) {
+    var fileBytes = null;
+    try {
+      var fs = require('fs');
+      // read file from a local directory
+      fileBytes = fs.readFileSync(path.resolve(__dirname, SignTest1File));
+    } catch (ex) {
+      // handle error
+      console.log('Exception: ' + ex);
+    }
+
+    // create an envelope to be signed
+    var envDef = new docusign.EnvelopeDefinition();
+    envDef.emailSubject = 'Please Sign my Node SDK Envelope';
+    envDef.emailBlurb = 'Hello, Please sign my Node SDK Envelope.';
+
+    // add a document to the envelope
+    var doc = new docusign.Document();
+    var base64Doc = Buffer.from(fileBytes).toString('base64');
+    doc.documentBase64 = base64Doc;
+    doc.name = 'TestFile.pdf';
+    doc.documentId = '1';
+
+    var docs = [];
+    docs.push(doc);
+    envDef.documents = docs;
+
+    // Add a recipient to sign the document
+    var signer = new docusign.Signer();
+    signer.email = userName;
+    var name = 'Pat Developer';
+    signer.name = name;
+    signer.recipientId = '1';
+
+    // this value represents the client's unique identifier for the signer
+    var clientUserId = '2939';
+    signer.clientUserId = clientUserId;
+
+    // create a signHere tab somewhere on the document for the signer to sign
+    // default unit of measurement is pixels, can be mms, cms, inches also
+    var signHere = new docusign.SignHere();
+    signHere.documentId = '1';
+    signHere.pageNumber = '1';
+    signHere.recipientId = '1';
+    signHere.xPosition = '100';
+    signHere.yPosition = '100';
+
+    // can have multiple tabs, so need to add to envelope as a single element list
+    var signHereTabs = [];
+    signHereTabs.push(signHere);
+    var tabs = new docusign.Tabs();
+    tabs.signHereTabs = signHereTabs;
+    signer.tabs = tabs;
+
+    // Above causes issue
+    envDef.recipients = new docusign.Recipients();
+    envDef.recipients.signers = [];
+    envDef.recipients.signers.push(signer);
+
+    // make the envelope with "created" (draft) status
+    envDef.status = 'created';
+
+    var envelopesApi = new docusign.EnvelopesApi(apiClient);
+
+    envelopesApi.createEnvelope(accountId, { envelopeDefinition: envDef })
+      .then(function (viewUrl) {
+        if (viewUrl) {
```

Comment:
Try to prevent nested promises. You can **return the promise** and use a few **"then"** after it with only one catch.

You can rewrite this like
```
envelopesApi.createEnvelope(accountId, { envelopeDefinition: envDef })
    .then(function (viewUrl) {
        if (!viewUrl) {
            // reject or return;
        }
        return envelopesApi.createRecipientView(accountId, envelopeSummary.envelopeId, { recipientViewRequest: returnUrlRequest });
    }
    .then(function (viewUrl) {
        if (viewUrl) {
            console.log('ViewUrl is ' + JSON.stringify(viewUrl));
            done();
        }
    })
    .catch(function (error) {
        if (error) {
            return done(error);
        }
    });
```